site/server/app/com/fortysevendeg/exercises/persistence/domain/UserQueries.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * scala-exercises-server
+ * Copyright (C) 2015-2016 47 Degrees, LLC. <http://www.47deg.com>
+ */
+
+package com.fortysevendeg.exercises.persistence.domain
+
+import cats.data.Xor
+import shared.User
+import shapeless._
+import ops.record._
+
+object UserCreation {
+
+  // TODO: add proper case objects with errors when switching to Postgres
+  sealed trait CreationError
+
```

Comment:
We should make all root sealed objects in ADTs be abstract classes extending Product with Serializable